### PR TITLE
[WiP] LLVM code coverage support

### PR DESCRIPTION
**Pull request by klecki**
Separate Clang flags from Cuda host compiler flags.
Add coverage options to CMake.
Add "example" scripts.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,31 @@ option(BUILD_LMDB "Build LMDB readers" OFF)
 option(BUILD_TENSORFLOW "Build TensorFlow plugin" OFF)
 option(BUILD_JPEG_TURBO "Build with libjpeg-turbo" ON)
 option(BUILD_NVJPEG "Build with nvJPEG" ON)
+option(COVERAGE_SOURCE "Use LLVM Source-based Code Coverage build" OFF)
+option(COVERAGE_SANITIZER "Use LLVM Sanitizer-based Code Coverage build" OFF)
+option(USE_ADDITIONAL_CLANG_CXX_FLAGS "Use ADDITIONAL_CLANG_CXX_FLAGS variable that is passed only to Clang, do not propagete those flags to CUDA" OFF)
```
Comment (Kh4L) [inline CMakeLists.txt]:
```suggestion
option(USE_ADDITIONAL_CLANG_CXX_FLAGS "Use ADDITIONAL_CLANG_CXX_FLAGS variable that is passed only to Clang, do not propagate those flags to CUDA" OFF)
```

File: tools/coverage_sanitizer.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# Based on https://clang.llvm.org/docs/SanitizerCoverage.html
+
+mkdir -p build_coverage_sanitizer
+cd build_coverage_sanitizer
+cmake -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DBUILD_LMDB=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCOVERAGE_SANITIZER=ON ..
+
+# Run
+ASAN_OPTIONS=protect_shadow_gap=0:detect_odr_violation=0:coverage=1 ./dali/python/nvidia/dali/test/dali_test.bin
+# Symbolize
```
Comment (JanuszL) [inline tools/coverage_sanitizer.sh]: I think L0 python tests should be run as well.

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
Stale PR, only .cc files are covered, closing for now.

**Pull request closed without merging**